crates/vm/src/arch/execution_mode/metered/mod.rs

```diff
@@ -16,16 +21,17 @@ use crate::arch::{
 /// Check segment every 100 instructions.
 const SEGMENT_CHECK_INTERVAL: u64 = 100;
 
-// TODO(ayush): fix these values
-const MAX_TRACE_HEIGHT: u32 = (1 << 23) - 100;
-const MAX_TRACE_CELLS: usize = 2_000_000_000; // 2B
-const MAX_INTERACTIONS: usize = BabyBear::ORDER_U32 as usize;
+const DEFAULT_MAX_TRACE_HEIGHT: u32 = (1 << 23) - 100;
+const DEFAULT_MAX_CELLS: usize = 2_000_000_000; // 2B
+const DEFAULT_MAX_INTERACTIONS: usize = BabyBear::ORDER_U32 as usize;
 
 pub struct MeteredExecutionControl<'a> {
-    // Constants
     air_names: &'a [String],
     pub widths: &'a [usize],
     pub interactions: &'a [usize],
+    max_trace_height: u32,
```

Comment:
should make some kind of struct for this

Reply:
i'll create a `SegmentationLimits` struct